packages/rich-text/src/plugins/List/transforms/toggleList.spec.tsx

```diff
@@ -0,0 +1,395 @@
+  });
+
+  it('should turn a list to multiple p', () => {
+    const input = (
+      <editor>
+        <hul>
+          <hli>
+            <hp>
+              <anchor />1
+            </hp>
+          </hli>
+          <hli>
+            <hp>2</hp>
+          </hli>
+          <hli>
+            <hp>
+              3<focus />
+            </hp>
+          </hli>
+        </hul>
+      </editor>
+    ) as any as PlateEditor;
+
+    const expected = (
+      <editor>
+        <hp>1</hp>
+        <hp>2</hp>
+        <hp>3</hp>
+      </editor>
+    ) as any as PlateEditor;
+
+    const { editor } = createTestEditor({
+      input,
+    });
+
+    toggleList(editor, { type: BLOCKS.UL_LIST });
+
+    assertOutput({ editor, expected });
+  });
+});
+
+describe('toggle over', () => {
+  it('should toggle different list types', () => {
+    const input = (
+      <editor>
+        <hul>
+          <hli>
+            <hp>
+              1<cursor />
+            </hp>
+          </hli>
+        </hul>
+      </editor>
+    ) as any as PlateEditor;
+
+    const expected = (
+      <editor>
+        <hol>
+          <hli>
+            <hp>1</hp>
+          </hli>
+        </hol>
+        <hp>
+          <htext />
+        </hp>
+      </editor>
+    ) as any as PlateEditor;
+
+    const { editor } = createTestEditor({
+      input,
+    });
+
+    toggleList(editor, { type: BLOCKS.OL_LIST });
+
+    assertOutput({ editor, expected });
+  });
+
+  it('should only toggle the nested list', () => {
+    const input = (
+      <editor>
+        <hul>
+          <hli>
+            <hp>1</hp>
+            <hul>
+              <hli>
+                <hp>
+                  11
+                  <cursor />
+                </hp>
+              </hli>
+            </hul>
+          </hli>
+        </hul>
+      </editor>
+    ) as any as PlateEditor;
+
+    const expected = (
+      <editor>
+        <hul>
+          <hli>
+            <hp>1</hp>
+            <hol>
+              <hli>
+                <hp>11</hp>
+              </hli>
+            </hol>
+          </hli>
+        </hul>
+        <hp>
+          <htext />
+        </hp>
+      </editor>
+    ) as any as PlateEditor;
+
+    const { editor } = createTestEditor({
+      input,
+    });
+
+    toggleList(editor, { type: BLOCKS.OL_LIST });
+
+    assertOutput({ editor, expected });
+  });
+
+  it('should only toggle everything that is selected', () => {
+    const input = (
+      <editor>
+        <hul>
+          <hli>
+            <hp>
+              <anchor />1
+            </hp>
+            <hul>
+              <hli>
+                <hp>
+                  11
+                  <focus />
+                </hp>
+              </hli>
+            </hul>
+          </hli>
+        </hul>
+      </editor>
+    ) as any as PlateEditor;
+
+    const expected = (
+      <editor>
+        <hol>
+          <hli>
+            <hp>1</hp>
+            <hol>
+              <hli>
+                <hp>11</hp>
```

Comment:
This one too, what happens to the focus?

Reply:
fixed